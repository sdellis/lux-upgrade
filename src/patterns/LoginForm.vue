<template>
  <form id="app" novalidate="true">
    <alert v-if="alert" type="alert" :status="alertStatus" :autoclear=true>{{ alertMessage }}</alert>
    <input-text id="email" :errormessage="errormessageEmail" :value="emailValue" label="Email" @inputblur="validate($event)" placeholder="Write your text" />
    <input-text id="pwd" :errormessage="errormessagePwd" :value="pwdValue" label="Password" @inputblur="validate($event)" placeholder="Write your text" />
    <input-button v-on:system-alert="showAlert($event)"
      :customAlertEvent="{ 'alertStatus': 'success', 'alertMessage': 'Successfully logged in!'}"
      type="button"
      variation="solid" block>
        Login
    </input-button>
  </form>
</template>

<script>
/**
 * This example is a grouping of the _counter_ and the _svg-icon_ elements. It
 * demonstrates how to do simple state management in a template. When the counter total
 * reaches 5, the icon is replaced by a warning.
 *
 * More complex state management should be handled with Vuex.
 */
export default {
  name: "LoginForm",
  status: "prototype",
  release: "1.0.0",
  type: "Pattern",
  metaInfo: {
    title: "Form with validation",
    htmlAttrs: {
      lang: "en",
    },
  },
  data: function() {
    return {
      errormessageEmail: "",
      errormessagePwd: "",
      emailValue: "",
      pwdValue: "",
      alert: false,
      alertStatus: "info",
      alertMessage: "Testing",
    }
  },
  methods: {
    validate(field) {
      if (field.id == "email") {
        this.emailValue = field.value
        if (/(.+)@(.+){2,}\.(.+){2,}/.test(field.value)) {
          this.errormessageEmail = ""
        } else {
          this.errormessageEmail = "You need to supply an email."
        }
      }
      if (field.id == "pwd") {
        this.pwdValue = field.value
        if (!field.value.length) {
          this.errormessagePwd = "You need to supply a password."
        } else {
          this.errormessagePwd = ""
        }
      }
    },
    showAlert(values) {
      this.alertMessage = values.alertMessage
      this.alertStatus = values.alertStatus
      this.alert = true
    },
  },
}
</script>

<style lang="scss" scoped>
</style>

<docs>
  ```jsx
  <login-form></login-form>
  ```
</docs>
