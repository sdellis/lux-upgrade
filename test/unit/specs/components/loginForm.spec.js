import { createLocalVue, mount } from "@vue/test-utils"
import LoginForm from "@/patterns/LoginForm.vue"

// create an extended `Vue` constructor
const localVue = createLocalVue()

let wrapper

describe("LoginForm.vue", () => {
  beforeEach(() => {
    wrapper = mount(LoginForm, {
      localVue,
      stubs: ["input-text", "input-button", "alert"],
    })
  })

  it("should validate both email and password fields", () => {
    wrapper.vm.validate({ id: "email", value: "foo" })
    expect(wrapper.vm.errormessageEmail).toBe("You need to supply an email.")
    wrapper.vm.validate({ id: "email", value: "foo@bar.com" })
    expect(wrapper.vm.errormessageEmail).toBe("")
    wrapper.vm.validate({ id: "pwd", value: "" })
    expect(wrapper.vm.errormessagePwd).toBe("You need to supply a password.")
    wrapper.vm.validate({ id: "pwd", value: "foo" })
    expect(wrapper.vm.errormessagePwd).toBe("")
  })

  it("should show an alert component when showAlert is called", () => {
    expect(wrapper.vm.alert).toBe(false)
    wrapper.vm.showAlert({ alertMessage: "Success!", alertStatus: "success" })
    expect(wrapper.vm.alertMessage).toBe("Success!")
    expect(wrapper.vm.alertStatus).toBe("success")
    expect(wrapper.vm.alert).toBe(true)
  })

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
