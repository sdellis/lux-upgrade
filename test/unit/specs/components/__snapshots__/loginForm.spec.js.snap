// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm.vue has the expected html structure 1`] = `
<form
  id="app"
  novalidate="novalidate"
>
  <!---->
   
  <input-text-stub
    errormessage=""
    id="email"
    label="Email"
    placeholder="Write your text"
    value=""
  />
   
  <input-text-stub
    errormessage=""
    id="pwd"
    label="Password"
    placeholder="Write your text"
    value=""
  />
   
  <input-button-stub
    block=""
    customalertevent="[object Object]"
    type="button"
    variation="solid"
  >
    
      Login
  
  </input-button-stub>
</form>
`;
